src/view/TableSortedModel.java

```diff
@@ -0,0 +1,129 @@
+package view;
+
+import core.Folder;
+import core.Item;
+
+import javax.swing.table.AbstractTableModel;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Created by TANIA BARRIONEVO on 18/05/2016.
+ */
+public class TableSortedModel extends AbstractTableModel {
+
+    private List<Item> items;
+    private final String[] columnNames = {"<html><strong>Name</strong></html>",
+            "<html><strong>Ext</strong></html>", "<html><strong>Size</strong></html>",
+            "<html><strong>Date</strong></html>", "<html><strong>Attr</strong></html>"};
+    private final Class[] columnTypes = new Class[]{String.class, String.class, String.class, Date.class, String.class};
+    private Object[][] data;
+
+    public TableSortedModel(String workingPath) {
+        items = populateItems(workingPath);
+
+        populateDataTable();
+    }
+
+    private void populateDataTable() {
+        data = new Object[items.size()][columnNames.length];
+
+        for (int index = 0; index < items.size(); index++) {
+            Item item = items.get(index);
+            data[index] = new Object[] { item.getName(),
+                    item.getType(),
+                    item.getSize(),
+                    item.getLastModified(),
+                    item.getFilePermissions()};
+
+        }
+    }
+
+    /**
+     * Returns a list of files and folders that are located on an absolute path.
+     *
+     * @param   workingPath   an absolute path giving the base location of the files and folders.
+     * @return                the list of files an folders.
+     */
+    private List<Item> populateItems(String workingPath) {
+        List<Item> items = new ArrayList<Item>();
+
+        try {
+            // Get all the folders/files
+            File fileSystem = new File(workingPath);
```

Comment:
All this logic of getting the files and directories should be replaced by a method inside your classes of item, file or directory